pennylane/fourier/spectrum.py

```diff
@@ -0,0 +1,261 @@
+        import numpy as np
+        from pennylane.fourier import spectrum
+
+        n_layers = 2
+        n_qubits = 3
+        dev = qml.device("default.qubit", wires=n_qubits)
+
+        @qml.qnode(dev)
+        def circuit(x, w):
+            for l in range(n_layers):
+                for i in range(n_qubits):
+                    qml.RX(x[i], wires=0, id="x"+str(i))
+                    qml.Rot(w[l,i,0], w[l,i,1], w[l,i,2], wires=0)
+            qml.RZ(x[0], wires=0, id="x0")
+            return qml.expval(qml.PauliZ(wires=0))
+
+        x = np.array([1, 2, 3])
+        w = np.random.random((n_layers, n_qubits, 3))
+        res = spectrum(circuit)(x, w)
+
+        for inp, freqs in res.items():
+            print(f"{inp}: {freqs}")
+
+        >>> 'x0': [-3.0, -2.0, -1.0, 0.0, 1.0, 2.0, 3.0]
+        >>> 'x1': [-2.0, -1.0, 0.0, 1.0, 2.0]
+        >>> 'x2': [-2.0, -1.0, 0.0, 1.0, 2.0]
+
+    .. note::
+        While the Fourier spectrum usually does not depend
+        on trainable circuit parameters and the actual values of the inputs,
+        it may still change with arguments of the QNode that determine the architecture
+        of the circuit.
+
+    The input-encoding gates to consider can also be explicitly selected by using the
+    ``encoding_gates`` keyword argument:
+
+    .. code-block:: python
+
+        dev = qml.device("default.qubit", wires=1)
+
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.RX(x[0], wires=0, id="x0")
+            qml.PhaseShift(x[0], wires=0, id="x0")
+            qml.RX(x[1], wires=0, id="x1")
+            return qml.expval(qml.PauliZ(wires=0))
+
+        x = np.array([1, 2])
+        res = spectrum(circuit, encoding_gates=["x0"])(x)
+
+        for inp, freqs in res.items():
+            print(f"{inp}: {freqs}")
+
+        >>> 'x0': [-2.0, -1.0, 0.0, 1.0, 2.0]
+
+    .. note::
+        The `spectrum` function does not check if the result of the
+        circuit is an expectation, or if gates with the same `id`
+        take the same value in a given call of the function.
+
+    The `spectrum` function works in all interfaces:
+
+    .. code-block:: python
+
+        import tensorflow as tf
+
+        dev = qml.device("default.qubit", wires=1)
+
+        @qml.qnode(dev, interface='tf')
+        def circuit(x):
+            qml.RX(x[0], wires=0, id="x0")
+            qml.PhaseShift(x[1], wires=0, id="x1")
+            return qml.expval(qml.PauliZ(wires=0))
+
+        x = tf.constant([1, 2])
+        res = spectrum(circuit)(x)
+
+        for inp, freqs in res.items():
+            print(f"{inp}: {freqs}")
+
+        >>> 'x0': [-1.0, 0.0, 1.0]
+        >>> 'x1': [-1.0, 0.0, 1.0]
+
+    """
+
+    @wraps(qnode)
+    def wrapper(*args, **kwargs):
+        qnode.construct(args, kwargs)
+        tape = qnode.qtape
+
+        freqs = {}
+        for op in tape.operations:
+            id = op.id
+
+            # if the operator has no specific ID,
+            # move to the next
```

Comment:
```suggestion
            # if the operator has no specific ID, move to the next
```
This and a few other comments in this function can be merged into a single line. Maybe this is more of a stylistic choice, though, so unless we have a specific convention they can be left as-is.